src/vs/base/node/macAddress.ts

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { exec } from 'child_process';
+import { isWindows } from 'vs/base/common/platform';
+
+const cmdline = {
+	windows: 'getmac.exe',
+	unix: '/sbin/ifconfig -a || /sbin/ip link'
+};
+
+const invalidMacAddresses = [
+	'00:00:00:00:00:00',
+	'ff:ff:ff:ff:ff:ff',
+	'ac:de:48:00:11:22'
+];
+
+function validateMacAddress(candidate: string): boolean {
+	let tempCandidate = candidate.replace(/\-/g, ':').toLowerCase();
+	for (let invalidMacAddress of invalidMacAddresses) {
+		if (invalidMacAddress === tempCandidate) {
+			return false;
+		}
+	}
+
+	return true;
+}
+
+export function getMac(): Promise<string> {
+	return new Promise((resolve, reject) => {
+		try {
+			exec(isWindows ? cmdline.windows : cmdline.unix, { timeout: 10000 }, (err, stdout, stdin) => {
```

Comment:
I am not sure we can trust the `timeout` parameter. It makes sense to me to have it, but I would also have the timeout mechanism we used to have before where we would resolve the promise after 10s automatically instead of relying on the OS to do this.